### PR TITLE
Common Arcana: Commonize Mana Discerning Methods

### PR DESCRIPTION
**Pull request by jandersson**
This commonizes the methods in crossing-trainer used for determining
appropriate mana values when none are specified.

### CODE REVIEW

File: common-arcana.lic [hunk 1]
```diff
@@ -349,6 +349,64 @@ module DRCA
...
+    UserVars.discerns[data['abbrev']] = discern_data
+    data['mana'] = discern_data['mana']
+    data['cambrinth'] = discern_data['cambrinth']
+    data
+  end
+
+  def calculate_mana(min, more, discern_data, cyclic, settings)
+    total = min + more
+    total = (total * settings.prep_scaling_factor).floor
+    discern_data['mana'] = [(total / 4.0).ceil, min].max
```
Comment (rpherbig) [inline common-arcana.lic]: This used to be `(total / 5.0` - how do we feel about changing it?

**Comment by jandersson** [comment]
Updated to latest calculate mana method, sorry about that. The previous one was 5 months old

**Comment by rpherbig** [comment]
It looks good to me. Are you planning to make any more changes?

**Comment by jandersson** [comment]
I think I covered everything. Id like to test it out by having crossing-train use these methods. Will report back

**Comment by jandersson** [comment]
Its working

**Comment by jandersson** [comment]
Nevermind, It isn't. Checking out different branches seems to break symlinks on windows. Will need to push a few more commits

**Comment by jandersson** [comment]
Issues seem to be resolved now

**Comment by rpherbig** [comment]
```
2017-12-17 06:44:59 -0500:>;crossing-training
2017-12-17 06:44:59 -0500:--- Lich: crossing-training active.
2017-12-17 06:45:05 -0500:--- Lich: error: wrong number of arguments (1 for 2)
2017-12-17 06:45:05 -0500:	crossing-training:429:in `check_discern'
2017-12-17 06:45:05 -0500:	crossing-training:518:in `cast_spell'
2017-12-17 06:45:05 -0500:--- Lich: crossing-training has exited.
```

Apparently there's a conflict since the DRCA and script-specific methods have the same name, and the "wrong" one is picked by Ruby.

**Pull request merged**